minet/youtube/scrapers.py

```diff
@@ -118,6 +118,6 @@ def scrape_channel_id(channel_url):
         raise err
 
     soup = BeautifulSoup(response.data.decode("utf-8"), "lxml")
-    channel_id = soup.find("meta", {"itemprop": "channelId"}).get("content")
-
-    return channel_id
+    tag = soup.find("meta", {"itemprop": "channelId"})
+    if tag:
+        return tag.get("content")
```

Comment:
In the method that use the function down the line yes, probably so. But in this function particularly I feel returning `None` is okay since it is not exposed to the users (or so I think).

Reply:
@Yomguithereal  is right. This function is not exposed to the user. Minet primarily uses `scrape_channel_id()` for its youtube client, which the CLI actions use. The relevant CLI action is `minet youtube channel-videos` which is meant to "Retrieve metadata about all Youtube videos from one or many channel(s) using the API." 

(_Under the hood, this command extract[s] the channel id from the given url or scrape[s] the website to find it if necessary. Then the command uses the API to retrieve information about videos stored in the main playlist of the channel supposed to contain all the channel's videos._)

However, the CLI action seems to only work with Youtube channel pages, and does not scrape the channel ID from a video page. I ran into the bug when I tried to repurpose Minet's internal function `scrape_channel_id()` for scraping a channel ID from a video page. I think it might've only occured then because a video can be inaccessible in a certain region/county (and thus the HTML changed) but I think a channel page will always be accessible / never encounter this problem.

Reply:
Therefore, we might not need an Exception, since there is already an exception raised from the CLI action when a video page is input instead of a channel page. When you use `minet youtube channel-videos` on a video page, the error raised is `minet.youtube.exceptions.YouTubeInvalidChannelTargetError`.

Reply:
Actually, should the if condition be to check if there was a response, instead of `if err:` ?

```python
def scrape_channel_id(channel_url):
    err, response = request(channel_url, pool=YOUTUBE_SCRAPER_POOL)

    if response:
        soup = BeautifulSoup(response.data.decode("utf-8"), "lxml")
        tag = soup.find("meta", {"itemprop": "channelId"})
        if tag:
            return tag.get("content")
```

Reply:
I agree. I changed my mind.